src/Nethermind/Nethermind.Core/Extensions/Bytes.cs

```diff
@@ -23,23 +23,25 @@ namespace Nethermind.Core.Extensions
 {
     public static unsafe partial class Bytes
     {
-        public static readonly IEqualityComparer<byte[]> EqualityComparer = new BytesEqualityComparer();
+        private static readonly uint s_instanceRandom = (uint)System.Security.Cryptography.RandomNumberGenerator.GetInt32(int.MinValue, int.MaxValue);
+
+        public static readonly BytesEqualityComparer EqualityComparer = new BytesEqualityComparer();
         public static readonly IEqualityComparer<byte[]?> NullableEqualityComparer = new NullableBytesEqualityComparer();
         public static readonly ISpanEqualityComparer<byte> SpanEqualityComparer = new SpanBytesEqualityComparer();
         public static readonly BytesComparer Comparer = new();
         public static readonly ReadOnlyMemory<byte> ZeroByte = new byte[] { 0 };
         public static readonly ReadOnlyMemory<byte> OneByte = new byte[] { 1 };
 
-        private class BytesEqualityComparer : EqualityComparer<byte[]>
+        public class BytesEqualityComparer : EqualityComparer<byte[]>
```

Comment:
Does it need to be public?

Reply:
To show the type the the Jit and go via concerete type rather than interface